CMakeLists.txt

```diff
@@ -1,22 +1,42 @@
 cmake_minimum_required(VERSION 3.12)
 
-set(CMAKE_CXX_STANDARD 17)
-
-add_definitions(-D_CRT_SECURE_NO_WARNINGS /wd4244 /wd4267 /wd4101 /wd4018 /wd4305)
-
 project(loop-tempo-estimator)
+set(CMAKE_CXX_STANDARD 17)
 
+if (MSVC)
+    # Suppress specific MSVC warnings
+    add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+    add_compile_options(
+        /wd4244  # Conversion from 'type1' to 'type2', possible loss of data
+        /wd4267  # Conversion from 'size_t' to 'type', possible loss of data
+        /wd4101  # Unreferenced local variable
+        /wd4018  # Signed/unsigned mismatch
+        /wd4305  # Truncation from 'type1' to 'type2'
+    )
+elseif (CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
+    # Suppress equivalent GCC/Clang warnings
+    add_compile_options(
+        -D_CRT_SECURE_NO_WARNINGS  # Define macro (harmless on non-MSVC compilers)
+        -Wno-conversion            # Suppress warnings about conversions
+        -Wno-sign-conversion       # Suppress sign conversion warnings
+        -Wno-unused-variable       # Suppress unused variable warnings
+        -Wno-sign-compare          # Suppress signed/unsigned comparison warnings
+        -Wno-float-conversion      # Suppress float conversion warnings
+    )
+endif()
+
+# Rest of your CMakeLists.txt
 add_subdirectory(loop-tempo-estimator)
 
 add_subdirectory(Catch2)
-# add_subdirectory(mpg123) Doesn't have a CMakeLists.txt file, needs a bit more work. Consider the alternative minimp3.
+# add_subdirectory(mpg123)  # Uncomment if you add a CMakeLists.txt for mpg123
```

Comment:
Looks like minimp3 works fine ; you may get rid of this and remove the submodule.

Reply:
fixed!